File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/search/aggregations/bucket/RangeBucket.java
package fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class RangeBucket extends Bucket {

    private String key;
    private String toAsString;
    private String fromAsString;


    /**
     * Build a new <code>RangeBucket</code> from a <code>JsonObject</code>,
     * calling parent constructor.
     *
     * @param json the <code>JsonObject</code>
     */
    public RangeBucket(JsonObject json) {
        super(json);
        RangeBucketConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>RangeBucket</code> into a <code>JsonObject</code>,
     * calling parent <code>toJson</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = super.toJson();
        RangeBucketConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/search/aggregations/bucket/RangeBucket.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.search.aggregations.bucket

import fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.RangeBucket
import fr.myprysm.vertx.elasticsearch.action.search.aggregations.Aggregation

fun RangeBucket(
        aggregations: Map<String, fr.myprysm.vertx.elasticsearch.action.search.aggregations.Aggregation>? = null,
        docCount: Long? = null,
        fromAsString: String? = null,
        key: String? = null,
        toAsString: String? = null): RangeBucket = fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.RangeBucket().apply {

    if (aggregations != null) {
        this.setAggregations(aggregations)
    }
    if (docCount != null) {
        this.setDocCount(docCount)
    }
    if (fromAsString != null) {
        this.setFromAsString(fromAsString)
    }
    if (key != null) {
        this.setKey(key)
    }
    if (toAsString != null) {
        this.setToAsString(toAsString)
    }
}

